itis.playground/Pages/Game.xcplaygroundpage/Contents.swift

```diff
@@ -0,0 +1,238 @@
+import Foundation
+
+enum Fraction: String{
+    case water = "Water"
+    case fire = "Fire"
+    case earth = "Earth"
+    case air = "Air"
+}
+
+protocol Student{
+    var fraction: Fraction { get set }
+    
+    var isAlive: Bool { get set }
+    var HP: Int { get set }
+    var damage: Int { get set }
+    var cry: String { get set }
+    var name: String { get set }
+    init(name: String)
+}
+
+protocol Advantage {
+    var advantage: Int { get }
+    func getAdv() -> Void
+}
+
+protocol Disadvantage {
+    var disadvantage: Int { get }
+    func getDis() -> Void
+}
+
+class Arena {
+    
+    var firstTeam: Team
+    
+    var secondTeam: Team
+    
+    init(team1: Team, team2: Team) {
+        firstTeam = team1
+        secondTeam = team2
+    }
+    
+    func startBattle() {
+        while firstTeam.isAlive, secondTeam.isAlive {
+            print(firstTeam.students[0]?.cry ?? "")
+            secondTeam.getDamage(damage: firstTeam.fullDamage())
+            if secondTeam.isAlive {
+                print(secondTeam.students[0]?.cry ?? "")
+                firstTeam.getDamage(damage: secondTeam.fullDamage())
+            }
+        }
+    }
+}
+
+class Team {
+    var students: [Optional<Student>]
+    let fraction: Fraction
+    var teamHP = 0
+    var isAlive = true
+    
+    init(studs: [Student]) {
+        self.students = studs
+        fraction = studs[0].fraction
+        teamHP = commonHP()
+    }
+    
+    func fullDamage () -> Int {
+        var result = 0
+        for s in students {
+            result += s?.damage ?? 0
+        }
+        return result
+    }
+    
+    func commonHP() -> Int {
+        var result = 0
+        for s in students{
+            result+=s?.HP ?? 0
+        }
+        return result
+    }
+    
+    func getDamage(damage: Int) -> Bool {
```

Comment:
если вопрос о том может ли студент сделать урон то лучше назвать вот так 
```suggestion
    func isGetDamage(damage: Int) -> Bool {
```